nestkernel/connection.h

```diff
@@ -150,6 +150,18 @@ class Connection
   void set_status( const DictionaryDatum& d, ConnectorModel& cm );
 
   /**
+   * Check syn_spec dictionary for parameters that are not allowed with the
+   * given connection.
+   *
+   * Will issue warning or throw error if an illegal parameter is found. The
+   * method does nothing if no illegal parameter is found.
+   *
+   * @note Classes requiring checks need to override the function with their own
+   * implementation, as this base class implementation does not do anything.
+   */
+  void check_synapse_params( const DictionaryDatum& d ) const;
```

Comment:
Yes, the technical term is *vtable*: For the 4G kernel we worked hard to avoid virtual functions in `Connection` objects so that these objects do not contain a vtable-pointer, which would cost 8B per synapse; see Kunkel et al (2014) for details.